plugins/inputs/knx_listener/knx_helpers.go

```diff
@@ -0,0 +1,205 @@
+package knx_listener
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/vapourismo/knx-go/knx"
+	"github.com/vapourismo/knx-go/knx/cemi"
+	"github.com/vapourismo/knx-go/knx/dpt"
+)
+
+// Helper to abstract away router or tunnel interfaces
+type KNXInterface interface {
+	Inbound() <-chan knx.GroupEvent
+	Close()
+}
+
+func GetKNXInterface(mode string, address string) (client KNXInterface, err error) {
+	switch mode {
+	case "tunnel":
+		c, e := knx.NewGroupTunnel(address, knx.DefaultTunnelConfig)
+		client, err = &c, e
+	case "router":
+		c, e := knx.NewGroupRouter(address, knx.DefaultRouterConfig)
+		client, err = &c, e
+	case "dummy":
+		c, e := NewDummyInterface()
+		client, err = &c, e
+
+		data := generate_data()
+		go send_regularly(&c, 1000.0*time.Millisecond, data)
+	default:
+		client, err = nil, fmt.Errorf("invalid interface type: %s", mode)
+	}
+	return
+}
+
+func send_regularly(di *KNXDummyInterface, period time.Duration, data []knx.GroupEvent) {
+	idx := 0
+	for range time.Tick(period) {
+		di.Send(data[idx])
+		idx = (idx + 1) % len(data)
+	}
+}
+
+func generate_event(a string, d dpt.DatapointValue) knx.GroupEvent {
```

Comment:
go functions usually are `camelCase` instead of `snake_case`

Reply:
Done